crates/solver-service/src/apis/quote/generation.rs

```diff
@@ -787,10 +797,10 @@ impl QuoteGenerator {
 			originChainId: U256::from(input_chain_id),
 			expires: expiry,
 			fillDeadline: fill_deadline,
-			inputOracle: alloy_primitives::Address::from_slice(&selected_oracle.0),
+			inputOracle: alloy_primitives::Address::from_slice(&input_oracle.0),
 			inputs: vec![[input_token_u256, input_amount_u256]],
 			outputs: vec![SolMandateOutput {
-				oracle: zero_bytes32,
+				oracle: output_oracle_bytes32,
```

Comment:
Instead of hardcoding to 0, I'm setting to the output oracle we support for that route.